### PR TITLE
gsoap: disallow building tools when crossbuilding

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **gsoap//all**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by prince-chrismc — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by madebr** [comment]
@uilianries @jgsogo 
From the logs, it's unclear to me what build/configuration has failed.
All failed logs in https://github.com/conan-io/conan-center-index/pull/3615#issuecomment-751421884 look fine to me.
Unless I'm missing something...

File: recipes/gsoap/all/conanfile.py [hunk 1]
```diff
@@ -25,27 +26,38 @@ def _build_subfolder(self):
...
+        "with_ipv6": True,
+        "with_cookies": True,
+        "with_c_locale": True,
+    }
 
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if tools.cross_building(self.settings):
+            self.options.tools = False
```
Comment (dmn-star) [inline recipes/gsoap/all/conanfile.py:50]:
```suggestion

```

**Review by dmn-star — CHANGES_REQUESTED** [review]

**Comment by dmn-star** [comment]
The conditional option setting will cause the build to fail on CCI.

**Comment by memsharded** [comment]
I am trying to build and fix this.
It seems that the upstream source files have been removed, and only latest is available in https://sourceforge.net/projects/gsoap2/files/gsoap-2.8/

**Comment by jgsogo** [comment]
Hi! As you've seen in #5516 we have stored the sources for existing versions in a public repository. We understand that gSOAP sources will be periodically removed from Sourceforge so we will need to keep adding them to the repository. We are thinking (and working) on a better solution.

Please, fix conflicts and rebase/merge with `master`, time to move this PR ahead 🚀

**Pull request closed without merging**